src/VisualStudio/CSharp/Impl/LanguageService/CSharpHelpContextService.cs

```diff
@@ -23,568 +23,566 @@
-                    text = Keyword("yield");
+
+                case SyntaxKind.RequiredKeyword:
+                    text = Keyword("required");
                     return true;
             }
-
-            text = null;
-            return false;
-
-            static bool ModifiersContains(SyntaxToken token, SyntaxKind kind)
+        }
+        else if (token.ValueText is "notnull" or "unmanaged")
+        {
+            if (token.Parent is IdentifierNameSyntax { Parent: TypeConstraintSyntax { Parent: TypeParameterConstraintClauseSyntax } })
             {
-                return CSharpSyntaxFacts.Instance.GetModifiers(token.Parent).Any(t => t.IsKind(kind));
+                text = Keyword(token.ValueText);
+                return true;
             }
         }
 
-        private static bool TryGetTextForKeyword(SyntaxToken token, [NotNullWhen(true)] out string? text)
+        text = null;
+        return false;
+    }
+    private static bool TryGetTextForCombinationKeyword(SyntaxToken token, [NotNullWhen(true)] out string? text)
+    {
+        switch (token.Kind())
         {
-            if (token.IsKind(SyntaxKind.InKeyword))
-            {
-                if (token.GetAncestor<FromClauseSyntax>() != null)
-                {
-                    text = Keyword("from");
-                    return true;
-                }
+            case SyntaxKind.PrivateKeyword when ModifiersContains(token, SyntaxKind.ProtectedKeyword):
+            case SyntaxKind.ProtectedKeyword when ModifiersContains(token, SyntaxKind.PrivateKeyword):
+                text = Keyword("privateprotected");
+                return true;
 
-                if (token.GetAncestor<JoinClauseSyntax>() != null)
-                {
-                    text = Keyword("join");
-                    return true;
-                }
-            }
+            case SyntaxKind.ProtectedKeyword when ModifiersContains(token, SyntaxKind.InternalKeyword):
+            case SyntaxKind.InternalKeyword when ModifiersContains(token, SyntaxKind.ProtectedKeyword):
+                text = Keyword("protectedinternal");
+                return true;
 
-            if (token.IsKind(SyntaxKind.DefaultKeyword))
-            {
-                if (token.Parent is DefaultConstraintSyntax)
-                {
-                    text = Keyword("defaultconstraint");
-                    return true;
-                }
+            case SyntaxKind.UsingKeyword when token.Parent is UsingDirectiveSyntax:
+                text = token.GetNextToken().IsKind(SyntaxKind.StaticKeyword)
+                    ? Keyword("using-static")
+                    : Keyword("using");
+                return true;
+            case SyntaxKind.StaticKeyword when token.Parent is UsingDirectiveSyntax:
+                text = Keyword("using-static");
+                return true;
+            case SyntaxKind.GlobalKeyword when token.Parent is UsingDirectiveSyntax:
+                text = Keyword("global-using");
```

Comment:
This needs a corresponding PR on dotnet/docs I think.

Reply:
yes.  that will come after this.